File: obs_replaykit/dock.py
"""post-install sanity check that the ReplayKit browser HTML landed."""

from __future__ import annotations

from typing import Callable, Optional

from .config import DOCK_TARGET

LogFn = Optional[Callable[[str], None]]

__all__ = ("DOCK_TARGET", "verify_dock_install")


# canonical files we expect under DOCK_TARGET. anything else the walker copies is fine; this is just the sanity-check set.
_REQUIRED_FILES = ("controls.html", "controls_app.html", "clips.html")


def verify_dock_install(log: LogFn = None) -> int:
    """count how many of the canonical dock files landed. logs a warning for any missing one so an install failure shows up in apply output instead of as a confusing 'couldnt load that page' inside obs."""
    if not DOCK_TARGET.is_dir():
        if log:
            log(f"warn: {DOCK_TARGET} does not exist after install")
        return 0

    present = 0
    missing = []
    for name in _REQUIRED_FILES:
        if (DOCK_TARGET / name).is_file():
            present += 1
        else:
            missing.append(name)

    if missing and log:
        log(f"warn: dock install missing: {', '.join(missing)}")
    elif log:
        log(f"-> {DOCK_TARGET} ({present} file(s))")
    return present

File: ReplayKitSetup/Dock.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

namespace ReplayKitSetup
{
    // post-install sanity check that the replaykit browser html landed. ported from obs_replaykit/dock.py.
    public static class Dock
    {
        // canonical files we expect under DOCK_TARGET. anything else the walker copies is fine; this is just the sanity-check set.
        private static readonly string[] RequiredFiles = { "controls.html", "controls_app.html", "clips.html" };

        // count how many of the canonical dock files landed. logs a warning for any missing one so an install failure shows up in apply output instead of as a confusing "couldnt load that page" inside obs.
        public static int VerifyDockInstall(Action<string> log = null)
        {
            if (!Directory.Exists(Config.DOCK_TARGET))
            {
                log?.Invoke("warn: " + Config.DOCK_TARGET + " does not exist after install");
                return 0;
            }

            int present = 0;
            var missing = new List<string>();
            foreach (var name in RequiredFiles)
            {
                if (File.Exists(Path.Combine(Config.DOCK_TARGET, name))) present++;
                else missing.Add(name);
            }

            if (missing.Count > 0) log?.Invoke("warn: dock install missing: " + string.Join(", ", missing));
            else log?.Invoke("-> " + Config.DOCK_TARGET + $" ({present} file(s))");
            return present;
        }
    }
}
